### ISSUE TITLE
Github repo linking

### ISSUE THREAD

**Issue by Shuudoushi** [KSP-SpaceDock/SpaceDock#33]
Like here http://puu.sh/naCd9/f893db1e53.png and for auto pull/download/wtfe updates and the like.

**Comment by jkoritzinsky** [comment]
To make this work the best, it should integrate as a [webhook](https://developer.github.com/webhooks/) that gets notified every time a release is released.

**Comment by politas** [comment]
Only issue I can see is that Github doesn't have metadata for which KSP version is supported, so mod authors would need to update that on Spacedock afterwards.

**Comment by jkoritzinsky** [comment]
Pull data from a `.ckan`, `.netkan`, or `.version` file if it is present in the repository (possible to get via GH API) maybe?

**Comment by politas** [comment]
Where possible, sure. Mainly need to make sure that mod authors setting up the link are aware of the limitations/requirements/possibilities.

**Comment by GenPage** [comment]
@politas That is find, part of the GitHub integration would be to not publish the mod, so that the modder will have to publish it manually first thus updating the mod on SpaceDock afterwards.

@jkoritzinsky I'm also not opposed to a version file if present. We'll need to flush the workflow out more

**Comment by Shuudoushi** [comment]
Version info could just be pulled from the tag that is created while drafting a release on github, as most modders use this tag to denote version number anyway.

**Comment by politas** [comment]
Mod version number, yeah, but not supported KSP version.

**Comment by Shuudoushi** [comment]
@politas That can be set during first time setup, i.g. where you create the mod page on spacedock itself.

**Comment by politas** [comment]
...Until a new version of KSP is released.

**Comment by inigmatus** [comment]
...new version nearly released. :)

**Comment by HebaruSan** [comment]
In case any modders are interested, Nertea has done this on the GitHub side via the API:

- https://github.com/post-kerbin-mining-corporation/build-deploy/blob/master/src/ksp_deploy/spacedock.py

**Comment by Maximilian-Nesslauer** [comment]
This Issue is quite old but fyi github workflows such as https://github.com/Maximilian-Nesslauer/KSA-AdvancedFlightComputer/blob/main/.github/workflows/spacedock-publish.yml work perfectly fine by using the API

**Comment by HebaruSan** [comment]
Good point, the advent of GitHub actions and workflows seems to have made any changes on the SpaceDock side unnecessary here.
